File: whole brain/load_tiff_slices.m
% Define inputs
filePaths = {
    "C:\Users\boba4\Box\Zach_repo\Projects\Remote_memory\PL-TeA wholebrain\ZZ216B\deeptrace_analysis\240510_ZZ216B_488_08-50-04\refined_model__zz_axon_20240708_nb8_st3_at15.tiff",
    "C:\Users\boba4\Box\Zach_repo\Projects\Remote_memory\PL-TeA wholebrain\ZZ215\deeptrace_analysis\240509_ZZ215_488_15-10-53\refined_model__zz_axon_20240708_nb8_st3_at15.tiff",
    "C:\Users\boba4\Box\Zach_repo\Projects\Remote_memory\PL-TeA wholebrain\ZZ293\deeptrace_analysis\241111_ZZ293_08x_488_09-48-51\refined_model__zz_axon_20240708_nb8_st3_at15.tiff",
    "C:\Users\boba4\Box\Zach_repo\Projects\Remote_memory\PL-TeA wholebrain\ZZ294\deeptrace_analysis\241111_ZZ294_08x_488_11-38-16\refined_model__zz_axon_20240708_nb8_st3_at15.tiff",
};
sliceNumber = 880; % Replace with your slice number
outputPath = "C:\Users\boba4\Box\Zach_repo\Projects\Remote_memory\PL-TeA wholebrain\batch\all\image making";

% Run the processing function
process_tiff_files(filePaths, sliceNumber, outputPath);


%% FXN %%
function process_tiff_files(filePaths, sliceNumber, outputPath)
    % Initialize an empty cell array to store slice pixel values from all files
    slices = {};

    for i = 1:length(filePaths)
        % Open the TIFF file
        tiffFile = filePaths{i};
        imgStack = loadtiff(tiffFile);

        % Reslice the stack with default parameters (transpose for MATLAB convention)
        reslicedStack = reslice_stack(imgStack);

        % Check if the slice number is within bounds
        if sliceNumber < 1 || sliceNumber > size(reslicedStack, 3)
            error('Slice number %d is out of range for file: %s', sliceNumber, tiffFile);
        end

        % Extract the desired slice
        sliceData = reslicedStack(:, :, sliceNumber);

        % Store the slice data in the cell array
        slices{i} = sliceData;
    end

    % Sum the matrices
    summedMatrix = sum(cat(3, slices{:}), 3);

    % Save the summed matrix as a TIFF file
    outputFile = fullfile(outputPath, 'summed_matrix.tif');
    saveastiff(summedMatrix, outputFile);

    % Display the summed matrix
    disp('Summed matrix:');
    disp(summedMatrix);
    fprintf('Saved summed matrix to: %s\n', outputFile);
end

function reslicedStack = reslice_stack(stack)
    % Placeholder for reslicing functionality
    % For simplicity, this function transposes the stack along the third dimension
    reslicedStack = permute(stack, [2, 1, 3]);
end

File: whole brain/load_tiff_slices.py
import imagej
import numpy as np
from pathlib import Path

def process_tiff_files(file_paths, slice_number, output_path):
    # Start an ImageJ instance
    ij = imagej.init()  # Assumes FIJI is installed and accessible

    # Initialize a list to store slice pixel values from all files
    slices = []

    for file_path in file_paths:
        # Open the TIFF file
        imp = ij.io().open(str(file_path))

        # Reslice with default parameters
        ij.py.run_macro("Reslice [/]", imp)
        resliced = ij.py.run_macro("run(\"Reslice\");", imp)

        # Convert the result to a NumPy array
        resliced_np = ij.py.to_numpy(resliced.getImage())

        # Extract the desired slice (adjusting for zero-based indexing)
        if slice_number < 1 or slice_number > resliced_np.shape[0]:
            raise ValueError(f"Slice number {slice_number} is out of range for file: {file_path}")
        slice_data = resliced_np[slice_number - 1, :, :]

        slices.append(slice_data)

    # Sum the matrices
    summed_matrix = np.sum(slices, axis=0)

    # Save the summed matrix as a TIFF file
    output_file = Path(output_path) / "summed_matrix.tif"
    ij.io().save(ij.dataset().create(summed_matrix), str(output_file))

    # Display the summed matrix
    print("Summed matrix:")
    print(summed_matrix)
    print(f"Saved summed matrix to: {output_file}")

# Define inputs
file_paths = [
    "C:\Users\boba4\Box\Zach_repo\Projects\Remote_memory\PL-TeA wholebrain\ZZ216B\deeptrace_analysis\240510_ZZ216B_488_08-50-04\refined_model__zz_axon_20240708_nb8_st3_at15.tiff",
    "C:\Users\boba4\Box\Zach_repo\Projects\Remote_memory\PL-TeA wholebrain\ZZ215\deeptrace_analysis\240509_ZZ215_488_15-10-53\refined_model__zz_axon_20240708_nb8_st3_at15.tiff",
    "C:\Users\boba4\Box\Zach_repo\Projects\Remote_memory\PL-TeA wholebrain\ZZ293\deeptrace_analysis\241111_ZZ293_08x_488_09-48-51\refined_model__zz_axon_20240708_nb8_st3_at15.tiff",
    "C:\Users\boba4\Box\Zach_repo\Projects\Remote_memory\PL-TeA wholebrain\ZZ294\deeptrace_analysis\241111_ZZ294_08x_488_11-38-16\refined_model__zz_axon_20240708_nb8_st3_at15.tiff",
]
slice_number = 5  # Replace with your slice number
output_path = "/path/to/output"  # Replace with your desired output directory

# Run the processing function
process_tiff_files(file_paths, slice_number, output_path)
